pkg/provider/aws/knowledgebase/api_gateway.go

```diff
@@ -131,11 +129,7 @@ var ApiGatewayKB = knowledgebase.Build(
 			listener.LoadBalancer.Scheme = "internal"
 			listener.DefaultActions = []*resources.LBAction{{TargetGroupArn: core.IaCValue{Resource: tg, Property: resources.ARN_IAC_VALUE}, Type: "forward"}}
 			listener.Port = 80
-			if isTgDest {
-				listener.Protocol = "TCP"
-			} else {
-				listener.Protocol = "HTTPS"
-			}
+			listener.Protocol = tg.Protocol
```

Comment:
Does this need to do something else if `isEc2Dest`?

Reply:
the target group protocol is set dependent on whether its ec2 or other so we should be good, but we also cant test since we dont know how we are running containers/code on ec2 so i cant be 100% sure